static/src/js/containers/SavedProjectsContainer/__tests__/SavedProjectsContainer.test.js

```diff
@@ -1,103 +1,259 @@
-        projects: {},
-        isLoading: false,
-        isLoaded: false
-      }
-    }
-
-    const expectedState = {
-      savedProjects: {},
-      savedProjectsIsLoading: false,
-      savedProjectsIsLoaded: false
+      authToken: 'mock-token'
     }
 
-    expect(mapStateToProps(store)).toEqual(expectedState)
+    expect(mapStateToProps(store)).toEqual({
+      authToken: 'mock-token',
+      earthdataEnvironment: 'prod-env'
+    })
   })
 })
 
-describe('SavedProjectsContainer component', () => {
-  describe('when passed the correct props', () => {
-    test('renders a table when a retrieval exists with one collection that has no title', () => {
-      const { enzymeWrapper } = setup({
-        savedProjects: [
-          {
-            id: '4517239960',
-            name: 'project 1',
-            path: '/search?p=!C123456-EDSC',
-            created_at: '2019-09-05 00:00:00.000'
-          },
-          {
-            id: '2057964173',
-            name: 'project 3',
-            path: '/search?p=!C123456-EDSC',
-            created_at: '2019-09-05 00:00:00.000'
-          }
-        ],
-        savedProjectsIsLoading: false,
-        savedProjectsIsLoaded: true,
-        onChangePath: jest.fn(),
-        onDeleteSavedProject: jest.fn(),
-        onFetchSavedProjects: jest.fn()
+describe('SavedProjectsContainer', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    SavedProjects.mockClear()
+    nock.cleanAll()
+    nock.disableNetConnect()
+  })
+
+  describe('When the component mounts with a valid authToken', () => {
+    test('fetches the projects and renders SavedProjects with projects', async () => {
+      const mockProjects = [
+        {
+          id: '123',
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      setup()
+
+      await waitFor(() => {
+        const callWithLoadedData = SavedProjects.mock.calls.find(
+          (call) => call[0].isLoaded === true
+                     && call[0].isLoading === false
+                     && call[0].projects.length > 0
+        )
+        expect(callWithLoadedData).toBeTruthy()
+
+        if (callWithLoadedData) {
+          const [propsInCall] = callWithLoadedData
+          expect(propsInCall.isLoaded).toBe(true)
+          expect(propsInCall.isLoading).toBe(false)
+          expect(propsInCall.projects).toEqual(mockProjects)
+          expect(typeof propsInCall.onDeleteProject).toBe('function')
+        }
+      })
+    })
+  })
+
+  describe('When the authToken is null', () => {
+    test('renders SavedProjects with no projects', async () => {
+      setup({
+        overrideProps: {
+          authToken: null
+        }
       })
 
-      expect(enzymeWrapper.find(SavedProjects).length).toBe(1)
+      await waitFor(() => {
+        expect(SavedProjects).toHaveBeenCalled()
```

Comment:
If you want to look at the last call use `toHaveBeenLastCalledWith({...})`
Also include `toHaveBeenCalledTimes()` to ensure SavedProjects is rendered the number of times we expect